extractor/filesystem/os/bazelmaven/bazel.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bazelmaven extracts maven packages from bazel build files.
+package bazelmaven
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path"
+	"strings"
+
+	"github.com/bazelbuild/buildtools/build"
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	bazelmavenmeta "github.com/google/osv-scalibr/extractor/filesystem/os/bazelmaven/metadata"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+)
+
+const (
+	// Name is the unique name of this extractor.
+	Name = "os/bazelmaven"
+)
+
+// Extractor extracts maven packages from bazel build files.
+type Extractor struct{}
+
+// New returns a new instance of the extractor.
+func New() filesystem.Extractor { return &Extractor{} }
+
+// Name of the extractor.
+func (e Extractor) Name() string { return Name }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities {
+	return &plugin.Capabilities{}
+}
+
+// FileRequired returns true if the file is a Bazel build file.
+func (e Extractor) FileRequired(api filesystem.FileAPI) bool {
+	return path.Base(api.Path()) == "BUILD.bazel" ||
```

Comment:
might be cleaner to do something like `slices.Contains({"BUILD.bazel", "MODULE.bazel", ...}, path.Base(api.Path()))`